src/Controllers/CMSExternalLinksController.php

```diff
@@ -2,63 +2,54 @@
 
 namespace SilverStripe\ExternalLinks\Controllers;
 
+use SilverStripe\Admin\AdminController;
+use SilverStripe\Control\HTTPResponse;
 use SilverStripe\ExternalLinks\Model\BrokenExternalPageTrackStatus;
 use SilverStripe\ExternalLinks\Jobs\CheckExternalLinksJob;
 use SilverStripe\ExternalLinks\Tasks\CheckExternalLinksTask;
-use SilverStripe\Control\Controller;
 use Symbiote\QueuedJobs\Services\QueuedJobService;
-use SilverStripe\Control\Middleware\HTTPCacheControlMiddleware;
 use SilverStripe\PolyExecution\PolyOutput;
-use SilverStripe\Security\Permission;
 
-class CMSExternalLinksController extends Controller
+class CMSExternalLinksController extends AdminController
 {
+    private static ?string $url_segment = 'externallinks';
+
+    private static string|array $required_permission_codes = [
+        'CMS_ACCESS_CMSMain',
+    ];
+
     private static $allowed_actions = [
         'getJobStatus',
         'start'
     ];
 
     /**
      * Respond to Ajax requests for info on a running job
-     *
-     * @return string JSON string detailing status of the job
      */
-    public function getJobStatus()
+    public function getJobStatus(): HTTPResponse
     {
-        if (!Permission::check('CMS_ACCESS_CMSMain')) {
-            return $this->httpError(403, 'You do not have permission to access this resource');
-        }
-        // Set headers
-        HTTPCacheControlMiddleware::singleton()->setMaxAge(0);
-        $this->response
-            ->addHeader('Content-Type', 'application/json')
-            ->addHeader('Content-Encoding', 'UTF-8')
```

Comment:
UTF-8 encoding is either not needed anymore or is done by default - works without that line.
The content type is set by `AdminController::jsonSuccess()`